### PR TITLE
python38, py38-tkinter, py38-htmldocs, py38-gdbm: update to 3.8.4

### PR DESCRIPTION
**Pull request by chrstphrchvz**
Add reminder to update `py38-htmldocs`

#### Description
https://docs.python.org/release/3.8.4/whatsnew/changelog.html#changelog

###### Type(s)

- [x] bugfix
- [x] enhancement
- [x] security fix

###### Tested on

macOS 10.15.5
Xcode command line tools 11.5

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [ ] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
Will squash on request
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL?

- [x] checked your Portfile with `port lint`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

File: lang/python38/files/patch-configure.diff [hunk 1]
```diff
@@ -13,7 +13,7 @@
    # has no effect, don't bother defining them
    Darwin/[6789].*)
      define_xopen_source=no;;
--  Darwin/1[0-9].*)
+-  Darwin/[12][0-9].*)
 +  Darwin/[1-9][0-9].*)
      define_xopen_source=no;;
    # On AIX 4 and 5.1, mbstate_t is defined only when _XOPEN_SOURCE == 500 but
```
Comment (chrstphrchvz) [inline lang/python38/files/patch-configure.diff:19]: Should this hunk be kept, or dropped in favor of upstream's change (see python/cpython#21113)? (I.e. do we really need to worry about Darwin 30 anytime soon, or wait until Darwin 100?)

Comment (reneeotten) [reply lang/python38/files/patch-configure.diff:19]: remove the patch and go with the upstream solution

Comment (chrstphrchvz) [reply lang/python38/files/patch-configure.diff:19]: Done in c0694eb1034

**Comment by chrstphrchvz** [comment]
I don't know why my forced-pushed commits aren't appearing here. I can see them on the branch in my fork: https://github.com/chrstphrchvz/macports-ports/commits/py-3.8.4

**Comment by chrstphrchvz** [comment]
Trying again in a new PR.

**Comment by chrstphrchvz** [comment]
#7751 has the new commits

**Pull request closed without merging**